app/models/miq_alert_status_action.rb

```diff
@@ -0,0 +1,30 @@
+class MiqAlertStatusAction < ApplicationRecord
+  ACTION_TYPES = %w(assign acknowledge comment unassign unacknowledge).freeze
+
+  belongs_to :miq_alert_status
+  belongs_to :assignee, :class_name => 'User'
+  belongs_to :user
+  validates :action_type, :acceptance => { :accept => ACTION_TYPES }, :presence => true
+  validates :user, :presence => true
+  validates :miq_alert_status, :presence => true
+  validates :comment, :presence => true, :if => "action_type == 'comment'"
+  validates :assignee, :presence => true, :if => "action_type == 'assign'"
+  validates :assignee, :absence => true, :unless => "action_type == 'assign'"
+  validate :only_assignee_can_acknowledge
+
+  after_save :update_status_acknowledgement
```

Comment:
@kbrock this saves a column on the owner of this action. Does it make sense to do it this way